File: js_validation/pardox/src/_ops/cluster.js
'use strict';

const { getLib } = require('../ffi');

module.exports = {
    /**
     * Scatter this DataFrame across cluster partitions.
     * @param {number} partitions  Number of partitions.
     * @returns {Object|null} Scatter metadata.
     */
    clusterScatter(partitions) {
        const lib = getLib();
        const result = lib.pardox_cluster_scatter(this._ptr, partitions);
        return result ? JSON.parse(result) : null;
    },
};

File: py_validation/pardox/_ops/cluster.py
from ..wrapper import lib


class ClusterMixin:

    def cluster_scatter(self, connection, name: str):
        """
        Scatter this DataFrame to a cluster connection.

        Args:
            connection: Cluster connection handle (from pdx.cluster_connect()).
            name (str): Dataset name to register on the cluster.

        Returns:
            DataFrame: Result pointer from scatter operation.
        """
        if not hasattr(lib, 'pardox_cluster_scatter'):
            raise NotImplementedError("pardox_cluster_scatter not found in Core.")
        ptr = lib.pardox_cluster_scatter(connection._ptr, name.encode('utf-8'), self._ptr)
        if not ptr:
            raise RuntimeError("cluster_scatter returned null.")
        return self.__class__._from_ptr(ptr)
